addons/mail/models/mail_link_preview_message.py

```diff
@@ -0,0 +1,48 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+from odoo.addons.mail.tools.discuss import Store
+
+
+class LinkPreviewMessage(models.Model):
+    _name = "mail.link.preview.message"
+    _inherit = "bus.listener.mixin"
+    _description = "Link between link previews and messages"
+
+    link_preview_id = fields.Many2one("mail.link.preview", index=True)
+    message_id = fields.Many2one("mail.message")
+    is_hidden = fields.Boolean(default=False)
+
+    def init(self):
+        self.env.cr.execute("CREATE UNIQUE INDEX IF NOT EXISTS mail_link_preview_id_message_id ON %s (message_id, link_preview_id)" % self._table)
```

Comment:
Why using this custom index ? It seems we will always go from fetched messages to link previews (as we fetch messages and then their side records), then message_id index is sufficient ?

Reply:
This index is a UNIQUE one, the goal is to avoid creating multiple instances of the same message_id, link_preview_id combination.

Reply:
Oh right, did not see the unique. But then do you need another index for link_preview_id ? Won't sql use the combined one anyway in some way ?

Reply:
And if we keep it, can it be migrated to the new models.UniqueIndex api ?